commons-logs/src/main/java/eu/europeana/api/commons/logs/LoggableDispatcherServlet.java

```diff
@@ -0,0 +1,213 @@
+package eu.europeana.api.commons.logs;
+
+import com.maxmind.geoip2.exception.AddressNotFoundException;
+import com.maxmind.geoip2.exception.GeoIp2Exception;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.springframework.web.servlet.DispatcherServlet;
+import org.springframework.web.servlet.HandlerExecutionChain;
+import org.springframework.web.util.ContentCachingRequestWrapper;
+import org.springframework.web.util.ContentCachingResponseWrapper;
+import org.springframework.web.util.WebUtils;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.Locale;
+import java.util.TimeZone;
+
+/**
+ * Created by Srishti on 14 September 2020
+ */
+public class LoggableDispatcherServlet extends DispatcherServlet {
+
+    private static final Logger LOG = LogManager.getLogger(LoggableDispatcherServlet.class);
+    private static final int BUFFERLENGTH = 5120;
+
+    @Override
+    protected void doDispatch(HttpServletRequest request, HttpServletResponse response) throws Exception {
+        long startTime = System.currentTimeMillis();
+        String serverTime = getCurrentDateTime();
+
+        if (!(request instanceof ContentCachingRequestWrapper)) {
+            request = new ContentCachingRequestWrapper(request);
+        }
+        if (!(response instanceof ContentCachingResponseWrapper)) {
+            response = new ContentCachingResponseWrapper(response);
+        }
+        HandlerExecutionChain handler = null;
+        try {
+            handler = getHandler(request);
+        } catch (Exception e) {
+            LOG.error("Exception occurred while retrieving HandlerExecutionChain from HttpServletRequest", e);
+        }
+
+        try {
+            super.doDispatch(request, response);
+        } catch (Exception e) {
+            LOG.error("Exception occurred while invoking DispatcherServlet", e);
+        } finally {
+            long processTime = System.currentTimeMillis() - startTime;
+            logRequest(request, response, handler, processTime, serverTime);
+            updateResponse(response);
+        }
+    }
+
+    /**
+     * Forms the LogMessage
+     *
+     * @param requestToCache
+     * @param responseToCache
+     * @param handler
+     */
+    private void logRequest(HttpServletRequest requestToCache, HttpServletResponse responseToCache, HandlerExecutionChain handler,
+                            long processTime, String serverTime) {
+        LogMessage logMessage = new LogMessage();
+
+        // TODO Have to decide if we will still set this values here
+        logMessage.setApp_guid(""); // app_id
```

Comment:
This is important information to have!

Reply:
But I need to figure out how we can get this

Reply:
added now